docs/drafts/derive-serviceaccount.md

```diff
@@ -0,0 +1,354 @@
+as well as a [`ClusterServiceVersion`](https://olm.operatorframework.io/docs/concepts/crds/clusterserviceversion/) (CSV) that describes the extension and its service account's permission requirements.
+
+The service account must have permissions to:
+ - create and manage the extension's `CustomResourceDefinition`s
+ - create and manage the resources packaged in the bundle
+ - grant the extension controller's service account the permissions it requires for its operation
+ - create and manage the extension controller's service account
+ - create and manage the `Role`s, `RoleBinding`s, `ClusterRole`s, and `ClusterRoleBinding`s associated with the extension controller's service account
+ - create and manage the extension controller's deployment
+
+Additionally, for clusters that use the [OwnerReferencesPermissionEnforcement](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#ownerreferencespermissionenforcement) admission plug-in, the service account must also have permissions to:
+ - update finalizers on the ClusterExtension to be able to set blockOwnerDeletion and ownerReferences
+
+It is good security practice to follow the [principle of least privilege(https://en.wikipedia.org/wiki/Principle_of_least_privilege)], and scope permissions to specific resource names, wherever possible.
+Keep in mind, that it is not possible to scope `create`, `list`, and `watch` permissions to specific resource names.
+
+Depending on the scope, each permission will need to be added to either a `ClusterRole` or a `Role` and then bound to the service account with a `ClusterRoleBinding` or a `RoleBinding`.
+
+### Example
+
+The following example illustrates the process of deriving the minimal RBAC required to install the [ArgoCD Operator](https://operatorhub.io/operator/argocd-operator) v0.6.0 provided by [OperatorHub.io](https://operatorhub.io/).
+The final permission set can be found in the [ClusterExtension sample manifest](../../config/samples/olm_v1alpha1_clusterextension.yaml) in the [samples](../../config/samples/olm_v1alpha1_clusterextension.yaml) directory.
+
+The bundle image for the ArgoCD Operator v0.6.0 can be sourced from [quay.io/operatorhubio/argocd-operator:v0.6.0](https://quay.io/operatorhubio/argocd-operator:v0.6.0).
+
+The bundle includes the following content:
+
+* `ClusterServiceVersion`:
+  - argocd-operator.v0.6.0.clusterserviceversion.yaml
+* `CustomResourceDefinition`s:
+  - argoproj.io_applicationsets.yaml
+  - argoproj.io_applications.yaml
+  - argoproj.io_appprojects.yaml
+  - argoproj.io_argocdexports.yaml
+  - argoproj.io_argocds.yaml
+* Additional resources:
+  - argocd-operator-controller-manager-service_v1_service.yaml
+  - argocd-operator-controller-manager-metrics-service_v1_service.yaml
+  - argocd-operator-manager-config_v1_configmap.yaml
+  - argocd-operator-metrics-reader_rbac.authorization.k8s.io_v1_clusterrole.yaml
```

Comment:
Doesn't have to be done as part of this PR since I know @perdasilva is working on some follow-ups for this PR regarding the actual bundle manifests, but should we make each of these file references links so that a reader can easily click it and open the raw YAML for the resource in a new tab?

Reply:
shall I make these changes @perdasilva ?

Reply:
does this read better @everettraven ?
<img width="739" alt="Screenshot 2024-09-30 at 11 32 36 PM" src="https://github.com/user-attachments/assets/0b0c0ef6-31c1-4c7e-8d6d-d2962f68b037">